### PR TITLE
Fire OnEditorEntityCreated notification in SandboxIntegrationManager::CreateNewEntityAtPosition

### PR DESCRIPTION
**Pull request by chiyenteng**
Bug fix for #6573 .

Currently if prefab system enabled, `SandboxIntegrationManager::CreateNewEntityAtPosition` will not fire `EditorEntityContextNotification::OnEditorEntityCreated` notification, causing the error in automated tests while setting prefab system enabled flag to true.

The test mentioned in #6573 with the repro steps taken passed  after the fix applied.

Tested using the following command:
`python\python.cmd -m pytest AutomatedTesting\Gem\PythonTests\largeworlds\landscape_canvas\TestSuite_Periodic.py::TestAutomation::test_LandscapeCanvas_AreaNodes_DependentComponentsAdded --build-directory windows_vs2019\bin\profile`

### CODE REVIEW

**Review by NicholasVanSickle — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp [hunk 1]
```diff
@@ -284,9 +271,11 @@ namespace AzToolsFramework
         {
             return;
         }
+
         SetupEditorEntity(entity);
```
Comment (srikappa-amzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp:275]: In all the 3 cases (PrefabPublicHandler::CreateEntity, EditorEntityContextComponent::CreateNewEditorEntity, EditorEntityContextComponent::CreateNewEditorEntity), SetupEditorEntity() will be called indirectly through HandleEntitiesAdded(). 
So, you will be doing the same operations twice for prefabs.
So, this line can also be put under if (m_isLegacySliceService)

**Review by srikappa-amzn — CHANGES_REQUESTED** [review]
Can you verify that this still remains fixed with this change? https://github.com/o3de/o3de/issues/6655

**Comment by chiyenteng** [comment]
> Can you verify that this still remains fixed with this change? #6655

Verified. No crash. Works as expected.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp (hunk 1 above)
Comment (chiyenteng) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp]: Fixed.

**Review by srikappa-amzn — APPROVED** [review]

**Pull request merged**